### PR TITLE
Transform flag cleanup

### PR DESCRIPTION
**Pull request by rbuckton**
With the recent change to add a `ContainsES2019` flag to the `TransformFlags` enum, we have now run out of space for additional flags. This PR removes a number of entries in the `TransformFlags` enum in favor of some simple checks. Benchmark comparisons against `master` indicate no noticeable regression in performance.

### CODE REVIEW

File: tests/baselines/reference/parseErrorIncorrectReturnToken.js [hunk 1]
```diff
@@ -17,9 +17,7 @@ let o = {
 string; // should be => not :
 // doesn't work in non-type contexts, where the return type is optional
 var f = function (n) { return function (string) { return n.toString(); }; };
-var o = {
```
Comment (weswigham) [inline tests/baselines/reference/parseErrorIncorrectReturnToken.js]: We didn't change the parser here, right? So why'd we lose the function in the output here?

File: tests/baselines/reference/strictModeInConstructor.js [hunk 2]
```diff
@@ -123,8 +123,8 @@ var Bs = /** @class */ (function (_super) {
 var Cs = /** @class */ (function (_super) {
     __extends(Cs, _super);
     function Cs() {
-        var _this = _super.call(this) || this;
         "use strict";
+        var _this = _super.call(this) || this;
```
Comment (weswigham) [inline tests/baselines/reference/strictModeInConstructor.js]: ❤️ drive by bugfix

File: tests/baselines/reference/parseErrorIncorrectReturnToken.js (hunk 1 above)
Comment (rbuckton) [reply tests/baselines/reference/parseErrorIncorrectReturnToken.js]: It was a parse error to begin with. Before the cleanup we would synthesize a body for a method without a body that wasn't an overload. As part of the cleanup we now always use the same code path to visit methods/functions even if the declaration itself doesn't have any changes. For correct code, nothing changes.

Comment (weswigham) [reply tests/baselines/reference/parseErrorIncorrectReturnToken.js]: Whereas now we elide it completely if there's no body and it has to have one? OK, I guess.

**Pull request merged**